testng-core/src/main/java/org/testng/internal/invokers/MethodRunner.java

```diff
@@ -129,7 +130,17 @@ public List<ITestResult> runInParallel(
       // testng387: increment the param index in the bag.
       parametersIndex += 1;
     }
-    PoolService<List<ITestResult>> ps = new PoolService<>(suite.getDataProviderThreadCount());
+
+    IObjectBag objectBag = context.getSuite().getObjectBag();
+    boolean sharedThreadPool = context.getSuite().getXmlSuite().isShareThreadPoolForDataProviders();
+
+    @SuppressWarnings("unchecked")
```

Comment:
Any suggestions on how to fix this? I am basically trying to figure out how to push in a `PoolService<List<ITestResult>>` using generics. I can only replace `PoolService` with `T`. I am not sure how to represent the `List<ITestResult>` here.

Reply:
Try replacing the diamond operators with the real value.
The inference is maybe not working as expected here.
And maybe it is a javac issue you should try to reproduce in its own independent context and write an issue ;)

Reply:
This is now removed.

Reply:
I'm still seing it 😉

Reply:
The `@SuppressWarnings` cannot be removed because the `ObjectBag` stores objects and I cannot make it store specific types. I did this because I hope to use the `ObjectBag` as a generic bean bag sort of place.